SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -88,18 +86,18 @@ void SpatialVirtualWorkerTranslator::AuthorityChanged(const Worker_AuthorityChan
 			QueryForWorkerEntities();
 		}
 	}
-	// TODO(zoning): If authority is received on the ACL component, we may need to update it.
 }
 
 // The translation schema is a list of Mappings, where each entry has a virtual and physical worker ID.
-// This method should only be called on workers who are not authoritative over the mapping.
-// TODO(harkness): Is this true? I think this will be called once the first time we get authority?
+// This method should only be called on workers who are not authoritative over the mapping and also when
+// a worker first becomes authoritative for the mapping.
 void SpatialVirtualWorkerTranslator::ApplyMappingFromSchema(Schema_Object* Object)
 {
 	if (NetDriver != nullptr &&
 		NetDriver->StaticComponentView->HasAuthority(SpatialConstants::INITIAL_VIRTUAL_WORKER_TRANSLATOR_ENTITY_ID, SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID))
 	{
 		UE_LOG(LogSpatialVirtualWorkerTranslator, Log, TEXT("(%s) ApplyMappingFromSchema called, but this worker is authoritative, ignoring"), *WorkerId);
+		return;
```

Comment:
For this if statement, in other places we would do something more like `check(NetDriver);` on the previous line, that would make the log more accurate?

Reply:
I figure this will also boil down to mocking the NetDriver in testing.

Reply:
yup, testing again.